examples/05.HTTP_SERVER/http_server.cc

```diff
@@ -0,0 +1,209 @@
+#include "timeout.h"
+#include <NetAPI.h>
+#include <debug.hh>
+#include <errno.h>
+#include <fail-simulator-on-error.h>
+#include <memory>
+#include <string_view>
+#include <thread.h>
+#include <tick_macros.h>
+
+using CHERI::Capability;
+
+using Debug            = ConditionalDebug<true, "HTTP server example test">;
+constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
+
+#define LISTEN_PORT 80
+DECLARE_AND_DEFINE_BIND_CAPABILITY(HTTPPort, UseIPv6, LISTEN_PORT);
```

Comment:
As part of an separate discussion David and I decided that the current situation is actually OK: The capability currently is either for v4 or v6, depending on the value of `CHERIOT_RTOS_OPTION_IPv6`. The idea is that if a client needs to listen on both it would create two sockets with two different bind capabilities. But since this is just an example I took `CHERIOT_RTOS_OPTION_IPv6` to make the cut whether or not to use a v6 or v4 socket.